src/track/cue.cpp

```diff
@@ -44,15 +45,52 @@ Cue::Cue(int id, TrackId trackId, Cue::Type type, double position, double length
           m_color(color) {
     DEBUG_ASSERT(!m_label.isNull());
     if (length) {
-        if (position != Cue::kNoPosition) {
+        if (position != mixxx::kNoPosition) {
             m_sampleEndPosition = position + length;
         } else {
             m_sampleEndPosition = length;
         }
     } else {
-        m_sampleEndPosition = Cue::kNoPosition;
+        m_sampleEndPosition = mixxx::kNoPosition;
     }
 }
+
+Cue::Cue(TrackId trackId, int sampleRateHz, const mixxx::CueInfo& cueInfo)
+        : m_bDirty(false),
+          m_iId(-1),
+          m_trackId(trackId),
+          m_type(cueInfo.getType()),
+          m_sampleStartPosition(mixxx::kNoPosition),
+          m_sampleEndPosition(mixxx::kNoPosition),
+          m_iHotCue(cueInfo.getHotCueNumber()),
+          m_label(cueInfo.getLabel()),
+          m_color(Color::kPredefinedColorsSet.noColor) {
+    DEBUG_ASSERT(!m_label.isNull());
+
+    const double sampleRateKhz = sampleRateHz / 1000.0;
+    const double millisecsToSamplesFactor = sampleRateKhz * mixxx::kEngineChannelCount;
+    DEBUG_ASSERT(millisecsToSamplesFactor > 0);
+
+    if (cueInfo.getStartPositionMillis() != mixxx::kNoPosition) {
+        m_sampleStartPosition = cueInfo.getStartPositionMillis() * millisecsToSamplesFactor;
+    }
+
+    if (cueInfo.getEndPositionMillis() != mixxx::kNoPosition) {
+        m_sampleEndPosition = cueInfo.getEndPositionMillis() * millisecsToSamplesFactor;
+    }
+
+    mixxx::RgbColor::optional_t color = cueInfo.getColor();
+    if (color) {
+        // All tracks have the same color
+        for (PredefinedColorPointer pColor : Color::kPredefinedColorsSet.allColors) {
+            if (mixxx::RgbColor(pColor->m_defaultRgba.rgb()) == *color) {
+                m_color = pColor;
+                break;
+            }
+        }
```

Comment:
Mismatching colors are silently ignored and get lost?

Reply:
The get mapped to `noColor`: https://github.com/mixxxdj/mixxx/pull/2523/files/de1e4cda42015d78d14f34bf775be8d7ad793fd6#diff-2300140c90b82c59481dabb496aeda5aR67

Reply:
I'll add a warning